Controllers/AutomationController.php

```diff
@@ -0,0 +1,479 @@
+    }
+
+    /***
+     * Generate filters for automation listing
+     *
+     * @param $startDate
+     * @param $endDate
+     * @param $automationType
+     * @param $executionStatus
+     * @param $search
+     * @param $hideInactive
+     *
+     * @return array
+     */
+    protected function generateFiltersForAutomationListing(
+    	$startDate,
+	    $endDate,
+	    $automationType,
+	    $executionStatus,
+        $search = '',
+        $hideInactive = ''
+    ) {
+        $filters = [];
+
+        //It is necessary, so no compromise on it.
+        $filters['automation_type'] = $automationType;
+
+        if(!empty($startDate)) {
+            $filters['start_date'] = $startDate;
+        }
+
+        if(!empty($endDate)) {
+            $filters['end_date'] = $endDate;
+        }
+
+        if(!empty($executionStatus)) {
+            $filters['execution_status'] = $executionStatus;
+        }
+
+        if(!empty($search)) {
+            $filters['search'] = $search;
+        }
+
+        if(!empty($hideInactive)) {
+            $filters['hideInactive'] = $hideInactive;
+        }
+
+        return $filters;
+    }
+
+    /***
+     * Enable disable automation
+     *
+     * @param Request $request
+     *
+     * @return array
+     * @throws Exception
+     */
+    public function enableDisableAutomation(Request $request)
+    {
+        $response = [
+            'status' => false,
+            'message' => 'Failed to enable/disable automation'
+        ];
+        try {
+            $automationId = $request->get('automation_id');
+            $status = $request->get('status')  == "true" ? true : false;
+            $isSuccessful = $this->_automationSchedulerRepository->toggleAutomationStatusByID($automationId, $status);
+            if($isSuccessful) {
+                $response['status'] = $isSuccessful;
+                $response['message'] = "Status updated successfully!";
+            }
+        } catch(Exception $exception) {
+            parent::log($exception, self::class);
+        }
+        return $response;
+    }
+
+	/**
+	 * @param $taskObject
+	 * @param string $query
+	 * @param integer $numRowsAffected
+	 * @param string $message
+	 */
+    public function sendSQLRanNotification($taskObject, $query, $numRowsAffected, $message = "") {
+    	$automationType = makeEnumValuePresentable($taskObject->automationType);
+
+        $notificationSeverity = NotificationSeverity::SUCCESS;
+	    
+	    if(empty($message)) {
+		    $message = "<b>Automation Type: $automationType ran</b><br><br><br>Following queries affected <b>$numRowsAffected</b> rows.<br><br><br><b>Queries:</b> <code_block>$query</code_block>";
+	    }
+
+	    NotificationUtility::getInstance()->sendNotification(
+		    $message,
+		    $notificationSeverity
+	    );
+    }
+    /***
+     * Purge redis keys
+     *
+     * @param $redisKeys
+     * @return string $redisKeysRemoveMessage
+     */
+    public function purgeRedisKeys($redisKeys)
```

Comment:
Normally separating backend functions into a separate class is better than having them in the controller